### PR TITLE
Fix #298 configure sirepo api-tornado

### LINKED ISSUE #298: Sirepo tornado
- Start N sirepo units with the same env, cmd, etc.
- environment in unit file has port which is only thing different
- environment cascades in docker_start via `-e` flag to docker run command
- use PartOf to create a single target to stop and start all server instances
- nginx config is simple proxy forwarding to multiple in a list (`upstream` is typical example)
- ports are assigned fixed
- alternative would be to use a systemd `@` (instance file) but this doesn't solve the docker_start environment cascade but might work.
- disable/stop `sirepo@*`, enable `sirepo@{n..m}` only if changes. Port range needs to be in a file (env would be fine as a comment)
- port collision
- script to enable just for convenience
- systemctl stop sirepo; systemctl disable sirepo; rm /etc/systemd/system/sirepo.service (or @ service)

### PR DESCRIPTION
**Pull request by robnagler**
Working at a basic level on a virtual machine. If sirepo.num_api_servers is set, api-tornado is configured.

- https://github.com/radiasoft/sirepo/issues/5659 needs to be investigated.
- static files need to be returned from nginx

### CODE REVIEW

**Review by e-carlin — CHANGES_REQUESTED** [review]
Code looks good. Two notes:

1. Why was some test data deleted?
2. I kept running into a server upgraded loop. To reproduce use the below 000.yml, go to srw, examples -> Light Source Facilities -> NSLS-II -> NSLS-II ESM beamline -> NSLS-II ESM beamline, go to the beamline tab, see a "server upgraded message". It doesn't happen 100% of the time so if it doesn't occur try again. It always happened after < 3 attempts for me.

```yaml
---
default:
  base_os:
    ip_forward: false
    volume_groups: {}
  base_users:
    add:
      - vagrant
    spec:
      joeblow:
        uid: 2002
        email: devnull@v4.radia.run
      marysmith:
        uid: 2003
        email: blackhole@v4.radia.run
        procmail_want_include: true
      vagrant:
        uid: 1000
        email: vagrant@v4.radia.run
        want_shell: true
  component:
    tls_crt_create: true
    tls_crt:
       misc_hosts:
         - alpha2.sirepo.com
         - bivio.biz.v5.radia.run
  db_bkp:
    max_copies: 2
    on_calendar: "6:30"
  docker:
    iptables: false
  docker_cache:
    host: ''
  docker_registry:
    host: v3.radia.run
  logrotate:
    verbose: true
    # sometime before bkp.on_calendar
    on_calendar: "7:30"
  network:
    devices: {}
    trusted: {}
  nginx:
    worker_processes: 1
  rsconf: {}
  rsconf_db:
    #TODO(robnagler) POSIT dev/nginx/nginx.conf.jinja
    # defunct elvin_server https://en.wikipedia.org/wiki/Elvin
    http_host: http://v3.radia.run:2916
    host_run_d: /srv
    root_u: root
    run_u: vagrant
    run_uid: 1000
    local_files:
      "/etc/base.rsconf":
        mode: 0o700
        owner: nobody
        group: vagrant
  sirepo:
    docker_image: radiasoft/sirepo
    auth:
      methods:
        - guest
      guest:
        expiry_days: 1
    job:
      max_message_bytes: 333m
    pkcli:
      service_processes: 1
      service_threads: 20
      service_port: 7000
      job_supervisor:
        port: 7004
  sirepo_jupyterhub:
    admin_users: [ 'vagrant' ]
    docker_image: radiasoft/jupyterhub
    jupyter_docker_image: radiasoft/beamsim-jupyter
    api_port: 7914
    configurable_http_proxy_api_port: 8112
    configurable_http_proxy_port: 8111
    uri_root: 'jupyter'

channel:
  dev:
    network:
      trusted:
        # v*.radia.run private network
        10.10.10.0/24:
          gateway: 10.10.10.2
        # random network to test
        216.17.132.32/27:
          gateway: 216.17.132.62
          search: radia.run
          nameservers: [ 162.159.27.72, 162.159.24.39 ]
      untrusted: {}

host:
  dev:
    v3.radia.run:
      base_os:
        volume_groups:
          docker:
            logical_volumes:
              docker:
                #POSIT: download/installers/vagrant-dev:config.persistent_storage.size=102400
                gigabytes: 99
                mount_d: /srv/docker
      docker:
        tls_host: v3.radia.run
      rsconf_db:
        components:
          - docker
          - rsconf
          - docker_registry
    v4.radia.run:
      base_users:
        add:
          - joeblow
          - marysmith
        spec:
          joeblow:
            want_shell: true
      docker:
        tls_host: v4.radia.run
      rsconf_db:
        components: [ sirepo, sirepo_job_supervisor ]
      sirepo:
        job_driver:
          docker:
            dev_volumes: false
            hosts: [ v4.radia.run ]
        num_api_servers: 3
        vhost: sirepo.v4.radia.run
```

File: rsconf/component/sirepo.py [hunk 1]
```diff
@@ -34,120 +34,168 @@ def internal_build_compile(self):
...
-                            port=8001,
                         ),
                     ),
-                ),
+                )
             )
-        )
+
+        def _tornado(jc, z):
+            z._first_port = int(z.pkunchecked_nested_get("pkcli.service.port"))
```
Comment (e-carlin) [inline rsconf/component/sirepo.py:115]: `pkcli.service_port`

Comment (robnagler) [reply rsconf/component/sirepo.py]: c0174efa60d2fff723c030b60aa568a89a338dd3

**Comment by robnagler** [comment]
> Why was some test data deleted?

1.in and 1-fconf.in were identical (except for minor differences which [I added to 1.in](https://github.com/radiasoft/rsconf/pull/302/files#diff-29986c792113af615ca8262663fd9163862c3782a7a95a458245de163ba404ceR419)). They were different during the migration from old yaml to fconf. Maintaining two datasets that were so comprehensive was slowing me down.

>  kept running into a server upgraded loop.

https://github.com/radiasoft/sirepo/issues/5675

**Comment by robnagler** [comment]
We can merge this without turning it on, if that's ok with you.

**Review by e-carlin — APPROVED** [review]

**Pull request merged**